crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -570,6 +570,42 @@ where
     }
 }
 
+/// The [`Command`] type for unregistering one-shot systems from [`Commands`](crate::system::Commands).
+pub struct UnregisterSystemCached<I, O, M, S>
+where
+    I: SystemInput + 'static,
+    S: IntoSystem<I, O, M> + Send + 'static,
+{
+    system: S,
+    _phantom: PhantomData<(fn() -> I, fn() -> O, fn() -> M)>,
```

Comment:
```suggestion
    _phantom: PhantomData<fn() -> (I, O, M)>,
```
Optional

Reply:
Another option is to remove the `IntoSystem` constraint from the type entirely and just have 

```rust
pub struct UnregisterSystemCached<S> {
    system: S,
}
```

You need the constraints on `fn unregister_system_cached` to have the `Command` impl, but I don't think you actually need them on the `struct`.